### PR TITLE
[FEAT] war leaderboard

### PR DESCRIPTION
**Pull request by rickfurt**
# Description

This is a working in progress, it will implement the war leaderboard. 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

# Checklist:

- [x] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [x] Screenshot if it includes any UI changes
- [x] I have read the contributing guidelines and agree to the T&Cs
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

Screen Shot 2022-09-06 at 2 29 34 pm

https://user-images.githubusercontent.com/26861091/188547702-9bccdd37-367f-4d98-ab8a-1e03efa41d3d.mov

### CODE REVIEW

**Comment by rickfurt** [comment]
Opening a PR to receive a feedback of the UI, and what could be improved

**Comment by j923sneaks** [comment]
Looks awesome! Small suggestion to add back and close buttons 😁

**Comment by rickfurt** [comment]
Screen Shot 2022-09-06 at 3 04 20 pm

Updated UI

**Comment by c-py** [comment]
@rickfurt Hey mate!!!

Nice one! This looks so good! 

What would it look like if it was pinned to the top of the screen? I think it would be amazing for all players to see the leaderboard while the war is going on ⚔️ 

Would it be possible to pull this data from on chain directly instead of via the API?

**Comment by rickfurt** [comment]
> @rickfurt Hey mate!!!
> 
> Nice one! This looks so good!
> 
> What would it look like if it was pinned to the top of the screen? I think it would be amazing for all players to see the leaderboard while the war is going on ⚔️
> 
> Would it be possible to pull this data from on chain directly instead of via the API?

I was unsure that would be possible, |

Let me do some testing and digging tonight about the direct fetch from the chain directly, I'll let you know

**Comment by mGranTY** [comment]
> > @rickfurt Hey mate!!!
> > Nice one! This looks so good!
> > What would it look like if it was pinned to the top of the screen? I think it would be amazing for all players to see the leaderboard while the war is going on ⚔️
> > Would it be possible to pull this data from on chain directly instead of via the API?
> 
> I was unsure that would be possible,
> 
> Let me do some testing and digging tonight about the direct fetch from the chain, I'll let you know

use the inventory contract and fetch id 918 and 918 using the totalSupply method

**Comment by j923sneaks** [comment]
Re: Steve's [comment](https://discord.com/channels/880987707214544966/905567246452129873/1016669600206835752)

> I love it, I think the only thing o would want is it on the main screen and not hidden in a modal. I think during war it would be good to see it all the time 

Maybe we display progress bar on the map (w/o %) then retain the same flow so that minute details wont bloat the screen

**Comment by rickfurt** [comment]
## Mobile view 
https://user-images.githubusercontent.com/26861091/188675799-7d2f7d6b-acb4-4f6d-948f-6d356f18aa4f.mov

Screen Shot 2022-09-07 at 1 26 59 am
Screen Shot 2022-09-07 at 1 26 52 am

## Desktop View

https://user-images.githubusercontent.com/26861091/188675832-50c41699-08f4-4b5d-8b03-a22a0da15581.mov

Screen Shot 2022-09-07 at 1 25 36 am
Screen Shot 2022-09-07 at 1 25 29 am
Screen Shot 2022-09-07 at 1 25 16 am
Screen Shot 2022-09-07 at 1 25 09 am

**Comment by ShiningMinions** [comment]
This is awesome!

File: src/features/war/components/Leaderboard.tsx [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+            <img
+              src={maleGoblin}
+              alt=""
+              style={{
+                width: `${PIXEL_SCALE * 16}px`,
+                height: `${PIXEL_SCALE * 18}px`,
+              }}
+            />
+          </div>
+        </div>
```
Comment (c-py) [inline src/features/war/components/Leaderboard.tsx:160]: Looking so good!

**suggestion:** What do you think about removing the `width` or `height` on the humans and goblins? I think they are looking a little bit squashed when both `width` and `height` is specified. If only `width` or `height` is specified, then the other will scale with the right ratio.

In the example below I removed the `width`:

Before:


After:



```suggestion
        <div className="flex justify-between w-full px-8 mb-16">
          <div className={"flex h-fit"}>
            <img
              src={femaleHuman}
              style={{
                height: `${PIXEL_SCALE * 20}px`,
              }}
              alt=""
            />
            <img
              src={maleHuman}
              alt=""
              style={{
                height: `${PIXEL_SCALE * 20}px`,
              }}
            />
          </div>
          <div className="absolute w-14 left-1/2 -translate-x-1/2 top-11">
            <img
              src={sword}
              alt=""
              className="absolute"
              style={{
                width: "36px",
                left: "50%",
                top: "50%",
                transform: "translate(-50%, -50%)",
              }}
            />
            <img
              src={sword}
              alt=""
              className="absolute left-1/2 -translate-x-1/2 top-1"
              style={{
                width: "36px",
                left: "50%",
                top: "50%",
                transform: "translate(-50%, -50%) scaleX(-1)",
              }}
            />
          </div>
          <div
            className="flex items-end"
            style={{
              transform: "scaleX(-1)",
            }}
          >
            <img
              src={femaleGoblin}
              alt=""
              style={{
                height: `${PIXEL_SCALE * 18}px`,
              }}
            />
            <img
              src={maleGoblin}
              alt=""
              style={{
                height: `${PIXEL_SCALE * 18}px`,
              }}
            />
          </div>
        </div>
```

**Comment by rickfurt** [comment]
nice catch @c-py  thanks for that

**Review by c-py — APPROVED** [review]
Awesome @rickfurt !!

You're the man!

**Pull request merged**